### PR TITLE
Fix sampling impact parameter

### PR DESCRIPTION
**Pull request by victor-gonzalez**
For getting the proper fraction of the cross section when the requested minimum value of the impact parameter is not zero

### CODE REVIEW

**Comment by jbernhard** [comment]
Thanks for finding this error, Victor!

File: src/collider.cxx [hunk 1]
```diff
@@ -108,7 +108,7 @@ std::tuple<double, int> Collider::sample_collision() {
 
   do {
     // Sample b from P(b)db = 2*pi*b.
-    b = bmin_ + (bmax_ - bmin_) * std::sqrt(random::canonical<double>());
+    b = std::sqrt(bmin_ * bmin_ + (bmax_ * bmax_ - bmin_ * bmin_) * random::canonical<double>());
```
Comment (jbernhard) [inline src/collider.cxx:111]: I think I understand this, but just to confirm, the point here is to sample b^2 uniformly?

**Comment by jbernhard** [comment]
@morelandjs @sabass the tests failed but not because of the code, it's because the Travis configuration is woefully out of date.

File: src/collider.cxx (hunk 1 above)
Comment (victor-gonzalez) [reply src/collider.cxx:111]: Yes. Actually, the patch is not mine. I took it from `UrQMD` when the chosen option for impact parameter weighting `CTO5` is set to `1`, `random b from bmin to bmax, bdb weighted` which we used to use when generating events: `init.f L.421`

**Pull request merged**